apps/api/src/jobs/consumers/notifications/notification.consumer.ts

```diff
@@ -82,6 +82,16 @@ export abstract class NotificationConsumer {
           `Notification with ID ${id} was not sent correctly as per provider. Another attempt will be made to send the notification`,
         );
         this.logger.log('Provider response: ' + JSON.stringify(response.result));
+
+        // Check to prevent program to constantly keep checking for confirmation status
+        if (notification.retryCount >= this.maxRetryCount) {
+          notification.deliveryStatus = DeliveryStatus.FAILED;
+          await this.notificationRepository.save(notification);
```

Comment:
Since we have repository.save() method call already in the finally block, is this needed here?

Reply:
While testing, I saw that if I did not save the fail status to db here, the webhook call would again be repeated multiple times

Reply:
Retested it and saw that the above issue stopped repeating after letting api run idle for a few minutes so removed repository.save